File: pinyou_web/src/main/webapp/js/service/specificationService.js
app.service("specificationService",function($http){
    this.search=function(pageNum,pageSize,entity){
        return $http.post("../specification/search?pageNum="+pageNum+"&pageSize="+pageSize,entity);
    }
    this.add=function(entity){
        return $http.post("../specification/add",entity);
    }
    this.findOne=function(id){
        return $http.get("../specification/findOne?id="+id);
    }
    this.update=function(entity){
        return $http.post("../specification/update",entity);
    }
    this.dele=function(ids){
        return $http.get("../specification/delete?ids="+ids);
    }

    //查询规格下拉列表
    this.findSpecList = function () {
        return $http.get("../specification/findSpecList");
    }
})

File: pinyou_manager/pinyou_manager_service/src/main/java/cn/css/pinyou_manager/pinyou_manager_service/SpecificationService.java
package cn.css.pinyou_manager.pinyou_manager_service;


import cn.css.pinyou_dto.PageResult;
import cn.css.pinyou_dto.Specification;
import cn.css.pinyou_pojo.domain.TbSpecification;

import java.util.List;
import java.util.Map;

public interface SpecificationService {

    /**
     * 根据条件分页查询
     *
     * @param pageNum
     * @param pageSize
     * @param specification
     * @return
     */
    PageResult search(Integer pageNum, Integer pageSize, TbSpecification specification);

    /**
     * 新增规格
     *
     * @param vo
     */
    void add(Specification vo);

    /**
     * 根据规格ID查询规格信息
     *
     * @param id
     * @return
     */
    public Specification findOne(Long id);

    /**
     * 修改
     *
     * @param vo
     */
    void update(Specification vo);

    /**
     * 根据ID删除规格
     *
     * @param ids
     */
    void delete(Long[] ids);

    /**
     * 需求：查询规格值，进行下拉列表展示
     */
    public List<Map> findSpecList();

}
